packages/financial-templates-lib/src/price-feed/DefiPulseTotalTVLPriceFeed.js

```diff
@@ -0,0 +1,164 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { parseFixed } = require("@ethersproject/bignumber");
+
+// An implementation of PriceFeedInterface that uses DefiPulse Data api to retrieve prices.
+class DEFI_PULSE_TOTAL_TVL_PriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the DefiPulseTVL_ALLPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String} apiKey DeFiPulse Data API key. Note: these API keys are rate-limited.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests. 
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(
+    logger,
+    web3,
+    apiKey,
+    lookback,
+    networker,
+    getTime,
+    minTimeBetweenUpdates,
+    decimals = 18,
+
+  ) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.apiKey = apiKey;
+
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.toBN = this.web3.utils.toBN;
+
+    this.historicalPrices = [{timestamp:0, value:0}];
+
+    this.convertDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), decimals).toString());
+    };
+
+    this.removeValuesOlderThanLookback = () => {
+
+        const currentTime = this.getTime();
+
+        this.historicalPrices = this.historicalPrices.filter(priceObj => priceObj.timestamp > (currentTime - lookback) );
+
+    }
+
+
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    let closestTime = {timestamp:0, value:0};
+
+    //go through all values and find time that that is the largest and still less than 'time'
```

Comment:
nit: here (and in other places) it does not seem you've linted your code correctly. We normally have spaces after `//` in the comments. You should try running the linter with `yarn lint-fix` to address these formatting issues.

Reply:
Is there a config file with a preferred linter settings somewhere in the repo? Im getting Command "lint-fix" not found when running yarn lint-fix

Reply:
did you install all the packages within the repo by running `yarn` in the root of the repo? it should work fine once you have done such.

Reply:
Got it. Thanks